### PR TITLE
Add buffer for NetCDFWriter

### PR DESCRIPTION
**Pull request by ph-kev**
As a first step towards completing #13, this PR adds a `NetCDFBuffer` for buffering data that will eventually be written to the NetCDF file.

The default behavior is preserved by specifying a buffer size of 0 or 1. Otherwise, a buffer is used instead which buffer the writes to the NetCDF files.

TODO

- [ ] Add documentation
- [ ] Simplify buffer creation

### CODE REVIEW

File: src/netcdf_buffer.jl [hunk 1]
```diff
@@ -0,0 +1,226 @@
...
+    times,
+    time_range;
+    start_date = nothing,
+)
+    first_t_idx = first(time_range)
+    first_time = first(times)
+    nc[time_name][time_range] = times
+    if first_t_idx == 1
+        first_lower_time_bnd = ifelse(
+            first(times) isa Dates.Dates.AbstractDateTime,
```
Comment (nefrathenrici) [inline src/netcdf_buffer.jl]:
```suggestion
            first(times) isa Dates.AbstractDateTime,
```

File: src/netcdf_writer.jl [hunk 2]
```diff
@@ -546,30 +601,49 @@ NVTX.@annotate function write_field!(
...
+        buffer = get(diagnostic.output_writer.buffers, diagnostic, nothing)
+        if isnothing(buffer)
+            diagnostic.output_writer.buffers[diagnostic] = NetCDFBuffer(
+                nc,
+                var.short_name,
+                start_date,
+                t,
+                interpolated_field,
+                max_capacity = writer.buffer_max_capacity,
+            )
```
Comment (nefrathenrici) [inline src/netcdf_writer.jl]:
```suggestion
        if isnothing(buffer)
            buffer = NetCDFBuffer(nc, var.short_name, start_date, t, interpolated_field,
                                  max_capacity = writer.buffer_max_capacity)
            diagnostic.output_writer.buffers[diagnostic] = buffer
        else
            push!(buffer, t, interpolated_field)
        end
```
I think the buffer variable needs to be set if `isnothing(buffer)`

File: test/writers.jl [hunk 3]
```diff
@@ -549,3 +549,194 @@ end
               Dates.Second(div(2^53, 1000))
     end
 end
+
+@testset "NetCDFBuffer tests" begin
```
Comment (nefrathenrici) [inline test/writers.jl]: Could you add a test flushing an empty buffer?
